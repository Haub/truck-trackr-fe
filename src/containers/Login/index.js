import React, { Component } from "react";
import { NavLink, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import "./Login.css";
import { addUser, loadProfile, loadEvents } from "../../actions";
import * as helper from "../../utilities";

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      passwordOne: "",
      signUp: false,
      locationType: "",
      businessName: "",
      address: "",
      phoneNumber: "",
      contactName: "",
      foodType: "",
      logo: null
    };
  }

  handleKeyPress = event => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  handleSubmit = async e => {
    // e.preventDefault();
    const { email, passwordOne, signUp, locationType } = this.state;


    if (signUp) {
      this.props.firebase
        .doCreateUserWithEmailAndPassword(email, passwordOne)
        .then(async authUser => {
          let user = this.cleanUserForSignUp(authUser);
          sessionStorage.setItem("uid", authUser.user.uid);
          let result = await helper.createNewUser(user, locationType);
          this.props.addUser(result[0].data);
          this.props.loadProfile(result[0]);
          this.props.loadEvents(result[1].included)
          this.props.history.push("/profile");
        })
        .catch(error => {
          this.setState({ error });
        });
    } else {
      this.props.firebase
        .doSignInWithEmailAndPassword(email, passwordOne)
        .then(async authUser => {
          sessionStorage.setItem("uid", authUser.user.uid);
          const user = this.cleanUserForLogin(authUser, locationType);
          const result = await helper.loginUser(user);
          console.log(result);
          this.props.addUser(result[0].data);
          this.props.loadProfile(result[0]);
          this.props.loadEvents(result[1].included)
          this.props.history.push("/profile");
        })
        .catch(error => {
          this.setState({ error });
        });
    }
  };

  cleanUserForSignUp = authUser => {
    const {
      username,
      email,
      passwordOne,
      signUp,
      locationType,
      businessName,
      address,
      phoneNumber,
      contactName,
      foodType
    } = this.state;
    let user;
    if (locationType === "food_trucks") {
      user = {
        name: businessName,
        food_type: foodType,
        contact_name: contactName,
        phone: phoneNumber,
        email,
        website: username,
        uid: authUser.user.uid
      };
    } else {
      user = {
        name: businessName,
        address: address,
        contact_name: contactName,
        phone: phoneNumber,
        email,
        uid: authUser.user.uid
      };
    }
    return user;
  };

  cleanUserForLogin = (authUser, locationType) => {
    let user;
    if (locationType === "food_trucks") {
      user = {
        uid: authUser.user.uid,
        account_type: "food_truck"
      };
    } else {
      user = {
        uid: authUser.user.uid,
        account_type: "brewery"
      };
    }
    return user;
  };

  toggleSignUp = () => {
    let newSignUp = !this.state.signUp;
    this.setState({ signUp: newSignUp });
  };

  fileChangedHandler = event => {
    this.setState({
      logo: event.target.files[0]
    });
  };

  uploadHandler = () => {
    console.log(this.state.logo);

    // const formData = new FormData()
    // formData.append('myFile', this.state.logo, this.state.logo.name)
    // axios.post('my-domain.com/file-upload', formData)
  };

  render() {
    const { signUp, locationType } = this.state;
    let showFoodType;

    if (signUp && locationType === "food_trucks") {
      showFoodType = true;
    }

    return (
      <main className="login">
        <form className="login-form" onSubmit={this.handleSubmit}>
          <input
            className="email-input"
            name="email"
            type="email"
            placeholder="Email"
            onChange={this.handleKeyPress}
          />
          <input
            className="password-input"
            name="passwordOne"
            type="password"
            placeholder="Password"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "business-name-input" : "hidden"}
            name="businessName"
            placeholder="Business Name"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "address-input" : "hidden"}
            name="address"
            placeholder="Address/City Located"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "phone-number-input" : "hidden"}
            name="phoneNumber"
            placeholder="Phone Number"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "contact-name-input" : "hidden"}
            name="contactName"
            placeholder="Contact Name"
            onChange={this.handleKeyPress}
          />
          <input
            className={showFoodType ? "food-type-input" : "hidden"}
            name="foodType"
            placeholder="Food Type"
            onChange={this.handleKeyPress}
          />
          <p className={signUp ? "business-type" : "hidden"}>
            What Type of Business:
          </p>
          <div className="button-holder">
            <input
              className={"food-truck-radio"}
              type="radio"
              name="locationType"
              value="food_trucks"
              id="food-truck-button"
              onChange={this.handleKeyPress}
            />
            <label className={"food-truck-label"} htmlFor="food-truck-button">
              Food Truck
            </label>
            <input
              className={"brewery-radio"}
              type="radio"
              name="locationType"
              value="breweries"
              id="brewery-button"
              onChange={this.handleKeyPress}
            />
            <label className={"brewery-label"} htmlFor="brewery-button">
              Brewery
            </label>
          </div>
          {/* <input
            className={signUp ? "logo-input" : "hidden"}
            name="logo"
            placeholder="Upload logo"
            onChange={this.handleKeyPress}
            type="file"
          />
          <button
            className={signUp ? "logo-button" : "hidden"}
            onClick={this.uploadHandler}
          >
            Upload!
          </button> */}
          <NavLink
            to="/profile"
            className="signin-button"
            onClick={e => this.handleSubmit(e)}
          >
            {signUp ? "Sign Up" : "Sign In"}
          </NavLink>
        </form>
        <button className="signup-button" onClick={this.toggleSignUp}>
          {signUp
            ? "Actually I already have an account"
            : "First Time Here? Sign Up!"}
        </button>
      </main>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  addUser: user => dispatch(addUser(user)),
  loadProfile: currentPage => dispatch(loadProfile(currentPage)),
  loadEvents: events => dispatch(loadEvents(events)) 
});

const { object, func } = PropTypes;

Login.propTypes = {
  user: object,
  addUser: func
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(Login));
