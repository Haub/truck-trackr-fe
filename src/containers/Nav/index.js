import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './nav.css'

export const Nav = (props) => {

    return (
        <nav>
            
        </nav>
    )
}