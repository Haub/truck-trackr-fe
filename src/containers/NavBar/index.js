import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { loadFoodTrucks, loadBreweries, loadProfile } from '../../actions';
import { mockFoodTrucks, mockBreweries } from '../../tests/testMocks';
import * as helper from '../../utilities'
import './NavBar.css'

export class NavBar extends Component {
    constructor(props) {
        super(props)
        this.state = {

        }
    }

    loadData = async (event) => {
        const businessType = event.target.getAttribute('name')

       const result = helper.fetchBusinesses(businessType)
        if(businessType === 'breweries'){
            this.props.loadProfile({page: 'breweries'})
            this.props.loadBreweries(result.data)
        } else {
            this.props.loadProfile({page: 'food_trucks'})
            this.props.loadFoodTrucks(result.data)
        }
    }

    loadMockData = (event) => {
        const businessType = event.target.getAttribute('name')

        if(businessType === 'breweries') {
            this.props.loadBreweries(mockBreweries)
        } else if(businessType === 'food_trucks') {
            this.props.loadFoodTrucks(mockFoodTrucks)
        }
    }

    render(){
        return (
            <nav className={this.props.displayStatus ? 'display' : 'hidden-nav'}>
                <ul className='nav-controls'>
                    <li className='nav-control my-profile' className={this.props.displayStatus ? 'nav-control my-profile' : 'hidden'}>My Profile</li>
                    <NavLink to='/breweries' onClick={(e) => this.loadData(e)} name='breweries' className={this.props.displayStatus ? 'nav-control breweries' : 'hidden'}>Breweries</NavLink>
                    <NavLink to='/food_trucks' onClick={(e) => this.loadData(e)} name='food_trucks' className={this.props.displayStatus ? 'nav-control food-trucks' : 'hidden'}>Food Trucks</NavLink>
                    <input className={this.props.displayStatus ? 'nav-control search-input' : 'hidden'} placeholder='Search'/>
                </ul>
            </nav>
        )
    }
}

export const mapDispatchToProps = (dispatch) => ({
    loadBreweries: (breweries) => dispatch(loadBreweries(breweries)),
    loadFoodTrucks: (foodTrucks) => dispatch(loadFoodTrucks(foodTrucks)) ,
    loadProfile: (profile) => dispatch(loadProfile(profile))
});

export default connect(null, mapDispatchToProps)(NavBar);