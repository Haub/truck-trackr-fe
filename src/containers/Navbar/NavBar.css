nav {
    width: 200px;
    background-color: #6D3E48;
    height: 100vh;
    position: absolute;
    left:0px;
    top: 0px;
}

.nav-controls {
    margin-top: 100px;
    height: 400px;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-around;
}

.nav-control, .search-input {
    font-size: 1.7rem;
    padding: 8px;
    width: 100%;
}

.search-input {
    border:none;
}

.nav-control:hover {
    background-color: #88515d
}