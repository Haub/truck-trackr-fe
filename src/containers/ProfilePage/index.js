import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './ProfilePage.css';
import SignUpPage from '../../components/SignUpPage';
import { addUser } from '../../actions';
import { mockProfile } from '../../tests/testMocks';

export class ProfilePage extends Component {
    constructor(props){
        super(props)
    }

    componentDidMount(){
        // if(!Object.keys(this.props.currentPage).length){
        //     this.props.history.push('/')
        // }
        this.props.addUser(mockProfile)
    }



    render(){
        const { currentPage, user } = this.props;
        // const header = currentPage.replace('_', ' ').toUpperCase()
        // const lowerCaseBiz = header.toLowerCase();
        if(!Object.keys(currentPage).length && !user){
           return(
               <div>
                <h3>hello</h3>   
               </div>
           )
        } else if(currentPage && user.type === 'food_truck'){
            const { attributes } = this.props.user;
            return(
                <div className='parent-container'>
                    <h2 className='biz-name'>{attributes.name}</h2>
                    <div className='profile-container'>
                        <div className='biz-container'>
                          <h5 className='biz-title'>Food Type:</h5>
                          <h5 className='biz-info'>{attributes.food_type}</h5>

                          <h5 className='biz-title'>Phone Number:</h5>
                          <h5 className='biz-info'>{attributes.phone}</h5>

                          <h5 className='biz-title'>Contact Name:</h5>
                          <h5 className='biz-info'>{attributes.contact_name}</h5>

                          <h5 className='biz-title'>Email:</h5>
                          <h5 className='biz-info'>{attributes.email}</h5>

                          <h5 className='biz-title'>Website:</h5>
                          <a href={attributes.website} target='blank' className='biz-info'>{attributes.website}</a>
                        </div>

                        <div className='upcoming-events-container'>
                          <h3 className='upcoming-events-title'>UPCOMING EVENTS</h3>
                          <h4 className='upcoming-events-date'>12/12 <span className='upcoming-events-status'>Need a Booking</span></h4>
                          <h4 className='upcoming-events-date'>12/13 <span className='upcoming-events-status'>@ Cerebral</span></h4>
                          <h4 className='upcoming-events-date'>12/14 <span className='upcoming-events-status'>@ Diebolt </span></h4>
                          <h4 className='upcoming-events-date'>12/15 <span className='upcoming-events-status'>Need a Booking</span></h4> 
                          <h4 className='upcoming-events-date'>12/16 <span className='upcoming-events-status'>@ Call To Arms</span></h4>
                          <h4 className='upcoming-events-date'>12/17 <span className='upcoming-events-status'>@ Zuni Street Brewery </span></h4>
                          <h4 className='upcoming-events-date'>12/18 <span className='upcoming-events-status'>Need a Booking</span></h4> 
                        </div>
                    </div>
                </div>
            )
        } else if (currentPage && user.type === 'brewery'){
            const { attributes } = this.props.currentPage.data
            return(
                <div>
                    <h3>hi!!!!</h3>
                </div>
            )
        } else {
            return(
                <div>
                </div>
            )
        }
    }
}

export const mapStateToProps = (state) => ({
    currentPage: state.currentPage,
    user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
    addUser: (user) => dispatch(addUser(user))
})

export default connect(mapStateToProps, mapDispatchToProps)(ProfilePage)