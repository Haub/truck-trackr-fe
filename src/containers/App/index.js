import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { SignUpPage } from '../../components/SignUpPage';
import { BusinessContainer } from '../../components/BusinessContainer';
import './App.css';
import hamburger from '../../assets/hamburger.svg';
import ProfilePage from '../ProfilePage';
import NavBar from '../NavBar';
import truck from '../../assets/food-truck.png';
import barrel from '../../assets/barrel-icon-new.png';
import { withFirebase } from '../../components/Firebase';
import { loadProfile } from '../../actions'

export class App extends Component {
  constructor() {
    super()
    this.state = {
      login: true,
      navOpen: true,
    }
  }

  componentDidMount(){
    if(!this.state.login){
      this.props.history.push('/login')
    }
    if(!this.props.currentPage) {
      this.props.history.push('/')
    }
  }

  handleNavBar = (e) => {
    document.querySelector('.menu-trigger').classList.toggle('active')
    this.setState ({navOpen: !this.state.navOpen})
  }

  render() {
    let navBar;
    const { navOpen } = this.state;
    const { match } = this.props;

    return (
      <div className='main'>
        <header>
          <h1 className='main-title'><span><img src={ truck } className='truck-icon'/></span>TruckTrackr<span><img src={ barrel } className='barrel-icon'/></span></h1>
          <section>
            <a onClick={ this.handleNavBar} className="menu-trigger" id="menu05">
              <span></span>
              <span></span>
              <span></span>
            </a>
          </section>
          
        </header>

        <div className='content-holder'>
        { this.state.navOpen &&
          <NavBar history={this.props.history}/>
        }
        { !this.state.navOpen &&
          <div id="wrapper">
            <span id="openbtn"  onClick={this.handleNavBar}></span>
          </div>
        }
        <Route
          exact path='/login'
          render={ props => (
            <SignUpPage history={this.props.history}/>
          )
           }
        />
        <Route 
          exact path='/'
          component={ ProfilePage }
        />
        <Route 
          exact path='/breweries'
          render={ props => (
            <BusinessContainer 
              data={this.props.breweries} 
              changeCurrentPage={this.props.changeCurrentPage} 
              currentPage={this.props.currentPage.page}
              loadProfile={this.props.loadProfile}
              history={this.props.history}/>
          )}
        />
        <Route 
          exact path='/food_trucks'
          render={ props => (
            <BusinessContainer 
              data={this.props.foodTrucks} 
              changeCurrentPage={this.props.changeCurrentPage} 
              currentPage={this.props.currentPage.page}
              loadProfile={this.props.loadProfile}
              history={this.props.history}/>
          )}
        />
        <Route path='/business/:businessName' component={ ProfilePage } />
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  breweries: state.breweries,
  foodTrucks: state.foodTrucks,
  currentPage: state.currentPage
});

export const mapDispatchToProps = (dispatch) => ({
  loadProfile: (profile) => dispatch(loadProfile(profile))
})

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(withFirebase(App)));