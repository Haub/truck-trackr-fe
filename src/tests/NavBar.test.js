import React from 'react';
import { shallow, mountww } from 'enzyme';
import { NavBar } from '../containers/NavBar/';
import { mapDispatchToProps } from '../containers/NavBar';
import { mockBreweries, mockFoodTrucks, mockProfile } from './testMocks.js';


describe ('NAVBAR', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<NavBar currentPage={mockProfile} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when loadBreweries is invoked', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.loadBreweries(mockBreweries);
      expect(mockDispatch).toHaveBeenCalled();
    });

    it('should call dispatch when loadFoodTrucks is invoked', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.loadFoodTrucks(mockFoodTrucks);
      expect(mockDispatch).toHaveBeenCalled();
    });

    it('should call dispatch when loadProfile is invoked', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.loadProfile(mockProfile);
      expect(mockDispatch).toHaveBeenCalled();
    });

  });

})