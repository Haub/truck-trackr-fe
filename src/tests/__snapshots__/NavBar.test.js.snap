// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NAVBAR should match the snapshot 1`] = `
<nav
  className="hidden-nav"
>
  <ul
    className="nav-controls"
  >
    <li
      className="hidden"
    >
      My Profile
    </li>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="hidden"
      name="breweries"
      onClick={[Function]}
      to="/breweries"
    >
      Breweries
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="hidden"
      name="food_trucks"
      onClick={[Function]}
      to="/food_trucks"
    >
      Food Trucks
    </NavLink>
    <input
      className="hidden"
      placeholder="Search"
    />
  </ul>
</nav>
`;
